arcgis-ios-sdk-samples/Display information/Read symbols from a mobile style/ReadSymbolsFromMobileStyleViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright © 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+/// A view controller that manages the interface of the Read Symbols from a
+/// Mobile Style sample.
+class ReadSymbolsFromMobileStyleViewController: UIViewController {
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemap: .topographic())
+            mapView.graphicsOverlays.add(AGSGraphicsOverlay())
+            mapView.touchDelegate = self
+        }
+    }
+    
+    /// The view controller that manages the symbol to be added when the map
+    /// view is tapped.
+    let symbolViewController = UIStoryboard(name: "ReadSymbolsFromMobileStyle", bundle: nil).instantiateViewController(withIdentifier: "SymbolViewController") as! ReadSymbolsFromMobileStyleSymbolViewController
```

Comment:
I don't think so. How would that work?

Reply:
Not very well in this case, now that I dig in :) You'd create a custom Object instance in the storyboard and set it to `ReadSymbolsFromMobileStyleSymbolViewController` class…

![2019-05-06_10-26-57](https://user-images.githubusercontent.com/25374/57231892-7efbb900-6fe9-11e9-9c93-09afebabace8.png)

… but I see now that it's a bad idea.

I didn't realize you are persisting state in the view controller's `UITableView`. I guess in this case it's OK but seems like bad practice (keeping a VC around with its view not displayed, just to maintain the state of the contained table view). I would have used segues and unwind segues to avoid referencing the SymbolViewController at all, but then a new SymbolViewController is instantiated each time you display it and you need a way to re-establish the table view to match the symbol, so this suddenly makes reasonable sense short of introducing new structs to pass around.